FreeBSD shim build broken due to sign-conversion warning

While building the native shims on FreeBSD, I see the following warnings. These warnings do not appear on Linux.

```
/usr/home/jan/corefx/src/Native/System.Native/pal_mount.cpp:84:49: error: implicit conversion changes signedness: 'int64_t' (aka 'long') to 'unsigned long' [-Werror,-Wsign-conversion]
        mpi->AvailableFreeSpace = bsize * stats.f_bavail;
                                        ~ ~~~~~~^~~~~~~~
1 error generated.
--- System.Native/CMakeFiles/System.Native.dir/pal_mount.cpp.o ---
*** [System.Native/CMakeFiles/System.Native.dir/pal_mount.cpp.o] Error code 1

make[2]: stopped in /usr/home/jan/corefx/bin/obj/FreeBSD.x64.Debug/Native
/usr/home/jan/corefx/src/Native/System.Native/pal_process.cpp:246:23: error: implicit conversion changes signedness: 'uint64_t' (aka 'unsigned long') to 'rlim_t' (aka 'long') [-Werror,-Wsign-conversion]
    native.rlim_cur = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal.CurrentLimit);
                    ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/home/jan/corefx/src/Native/System.Native/pal_process.cpp:247:23: error: implicit conversion changes signedness: 'uint64_t' (aka 'unsigned long') to 'rlim_t' (aka 'long') [-Werror,-Wsign-conversion]
    native.rlim_max = ConvertFromManagedRLimitInfinityToPalIfNecessary(pal.MaximumLimit);
                    ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/home/jan/corefx/src/Native/System.Native/pal_process.cpp:253:83: error: implicit conversion changes signedness: 'const rlim_t' (aka 'const long') to 'uint64_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    pal.CurrentLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native.rlim_cur);
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~^~~~~~~~
/usr/home/jan/corefx/src/Native/System.Native/pal_process.cpp:254:83: error: implicit conversion changes signedness: 'const rlim_t' (aka 'const long') to 'uint64_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    pal.MaximumLimit = ConvertFromNativeRLimitInfinityToManagedIfNecessary(native.rlim_max);
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~^~~~~~~~
4 errors generated.
```

--- a/src/Native/System.Native/pal_mount.cpp
+++ b/src/Native/System.Native/pal_mount.cpp
@@ -71,14 +71,17 @@ extern "C" int32_t GetSpaceInfoForMountPoint(const char* name, MountPointInforma
     int result = statfs(name, &stats);
     if (result == 0)
     {
-        // Note that f_bsize has a signed integer type on some platforms but musn't be negative.
+        // Note that these have signed integer types on some platforms but musn't be negative.
         // Also, upcast here (some platforms have smaller types) to 64-bit before multiplying to
         // avoid overflow.
         uint64_t bsize = UnsignedCast(stats.f_bsize);
+        uint64_t bavail = UnsignedCast(stats.f_bavail);
+        uint64_t bfree = UnsignedCast(stats.f_bfree);
+        uint64_t blocks = UnsignedCast(stats.f_blocks);
 
-        mpi->AvailableFreeSpace = bsize * stats.f_bavail;
-        mpi->TotalFreeSpace = bsize * stats.f_bfree;
-        mpi->TotalSize = bsize * stats.f_blocks;
+        mpi->AvailableFreeSpace = bsize * bavail;
+        mpi->TotalFreeSpace = bsize * bfree;
+        mpi->TotalSize = bsize * blocks;
     }
     else
     {

--- a/src/Native/System.Native/pal_process.cpp
+++ b/src/Native/System.Native/pal_process.cpp
@@ -3,9 +3,11 @@
 
 #include "pal_config.h"
 #include "pal_process.h"
+#include "pal_utilities.h"
 
 #include <assert.h>
 #include <errno.h>
+#include <limits>
 #include <signal.h>
 #include <stdlib.h>
 #include <sys/resource.h>
@@ -214,23 +216,24 @@ static int32_t ConvertRLimitResourcesPalToPlatform(RLimitResources value)
 // Because RLIM_INFINITY is different per-platform, use the max value of a uint64 (which is RLIM_INFINITY on Ubuntu)
 // to signify RLIM_INIFINITY; on OS X, where RLIM_INFINITY is slightly lower, we'll translate it to the correct value
 // here.
-static uint64_t ConvertFromManagedRLimitInfinityToPalIfNecessary(uint64_t value)
+static rlim_t ConvertFromManagedRLimitInfinityToPalIfNecessary(uint64_t value)
 {
-    if (value == UINT64_MAX)
+    // rlim_t type can vary per platform, so we also treat anything outside its range as infinite.
+    if (value == UINT64_MAX || value > std::numeric_limits<rlim_t>::max())
         return RLIM_INFINITY;
-    else
-        return value;
+
+    return static_cast<rlim_t>(value);
 }
 
 // Because RLIM_INFINITY is different per-platform, use the max value of a uint64 (which is RLIM_INFINITY on Ubuntu)
 // to signify RLIM_INIFINITY; on OS X, where RLIM_INFINITY is slightly lower, we'll translate it to the correct value
 // here.
-static uint64_t ConvertFromNativeRLimitInfinityToManagedIfNecessary(uint64_t value)
+static uint64_t ConvertFromNativeRLimitInfinityToManagedIfNecessary(rlim_t value)
 {
     if (value == RLIM_INFINITY)
         return UINT64_MAX;
-    else
-        return value;
+
+    return UnsignedCast(value);
 }
 
 static void ConvertFromRLimitManagedToPal(const RLimit& pal, rlimit& native)

